docs/preview/yamlgettingstarted-templates.md

```diff
@@ -0,0 +1,295 @@
+# YAML getting started - Templates (not available, for discussion only)
+
+## Example: Simple input
+
+The following example illustrates two things:
+
+1. Front-matter is used to define the default values for inputs.
+
+2. Inputs are expanded using template expressions. A value that starts and ends
+   with `${{ }}` indicates a template expression.
+
+Template:
+
+```yaml
+---
+# Default values
+inputs:
+  solution: '**/*.sln'
+---
+steps:
+- task: msbuild@1
+  inputs:
+    solution: ${{ inputs.solution }}
+- task: vstest@2
+  inputs:
+    solution: ${{ inputs.solution }}
+```
+
+Consumer:
+
+```yaml
+steps:
+- template: steps/msbuild.yml
```

Comment:
How is a template versioned?  Where are templates read from?  How can I reference specific versions of a template?

Reply:
Sorry I realized afterward I did not make this clear with the PR. There is a lot of details I did not cover with this document. This document focuses on the details of the syntax. I'll try to hastily answer the other questions, but really I need to step back and figure out how to fit that info into this document.

We are planning for the templates to exist in a repo. The common case is templates in the same repo, so you specify the relative path to the file. It is expected the file would exist in the same commit.

We are also planning to support templates in another repo. Something like:

```yaml
resources:
  repositories:
  - repository: foo
    url: https://contoso.visualstudio.com/myproj/foo
    sourceVersion: master

steps:
- template: shared-steps.yml@foo
```

Reply:
Got it.  It is definitely required that they can live in another repo.  I also think that it would be good to allow them to live in a nuget package or something like it.  Packages provide a good way of versioning and deploying new templates.